packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -124,6 +124,9 @@ Future<void> _runCmake(String buildModeName, Directory sourceDir, Directory buil
   final bool needCrossBuildOptionsForArm64 = needCrossBuild
       && targetPlatform == TargetPlatform.linux_arm64;
   int result;
+  if (!globals.processManager.canRun('cmake')) {
+    throwToolExit(globals.userMessages.cmakeMissing);
```

Comment:
Do you want to remove the `on ArgumentError` block? Or at least make it throw the same thing?

Reply:
good thinking

Reply:
I removed the catch, as I'm assuming if we get an `ArgumentError` that means it's a tool bug.

Reply:
You can tell I wrote this code when I was less familiar with Dart. Otherwise the fact that I was catching an `Error` type in the first place would have been a big red flag that the code was wrong...